src/__tests__/addRef.coffee

```diff
@@ -14,8 +14,32 @@ Comp = flow(
     </div>
 )
 
+CompWithInitialValue = flow(
+  addRef 'inputRef', 'initial'
+  ({inputRef}) ->
+    <p>{inputRef.current}</p>
+)
+
+CompWithInitialValueFromProps = flow(
+  addProps value: 'initial'
+  addRef 'inputRef', ({value}) -> value
+  ({inputRef}) ->
+    <p>{inputRef.current}</p>
+)
+
 describe 'addRef', ->
-  test 'works', ->
-    {getByText, getByTestId} = render <Comp />
-    fireEvent.click getByText /update/
-    expect(document.activeElement).toBe getByTestId 'input'
+  describe 'with no initial value', ->
+    test 'works', ->
+      {getByText, getByTestId} = render <Comp />
+      fireEvent.click getByText /update/
+      expect(document.activeElement).toBe getByTestId 'input'
+  describe 'with initial value', ->
+    test 'works', ->
+      {getByText} = render <CompWithInitialValue />
+      getByText 'initial'
+      undefined
```

Comment:
I'm not sure what your usual approach would be here. (I assume not this.) Calling `getByText` is sufficient as a test, but I need to return `undefined` or a promise.

Reply:
Ya that's a fine workaround for Coffeescript's implicit return not playing nicely with a library that's finicky about return values

I guess you could override/wrap `test()` to sniff the real `test()`'s return value and only "forward" it if it's a promise

Reply:
I'm fine with it if it's not too weird.